### PR TITLE
srcflux merged fluxes and friends

### LINKED ISSUE #384: srcflux default energy band
Need to implement `band=default` ala `flux_obs` and family.    Use `broad` for ACIS and `wide` for HRC.

### PR DESCRIPTION
**Pull request by kglotfelty**
The code to compute merged fluxes has been around for a while but has not been enabled (commented out).  This update enables this code; so now when users supply multiple event files they will also get an estimate of the combined flux.

This update also includes the parameter change to set the energy band to default, and to add the random_seed parameter so we can control the simulate_psf (psfmethod=marx) output.

This fixes : #384   #381  #399 #414

### CODE REVIEW

**Comment by kglotfelty** [comment]
Need to add the src number to the merge report 

https://github.com/cxcsds/ciao-contrib/blob/e010ce6ffb79ae594073c7511f2e9d154436d961/bin/srcflux#L2469

**Comment by DougBurke** [comment]
I assume this is still not ready for release?

**Comment by kglotfelty** [comment]
Correct.  count-rates look good; need to get verification that model fluxes are consistent w/ csc.

File: bin/srcflux [hunk 1]
```diff
@@ -21,7 +21,7 @@
 from __future__ import print_function
```
Comment (DougBurke) [inline bin/srcflux:21]: This isn't needed anymore (as we've dropped python 2). Can you remove it?

File: bin/srcflux [hunk 2]
```diff
@@ -1903,21 +1900,44 @@ def merge_modelflux( stk_params, myparams ):
...
-        combine_spectra.src_arf=[x for x in arf_files if os.path.exists(x)]
+        rmf_files = [ p.outroot+"{:04d}.rmf".format(snum+1) for p in stk_params ]
+
+        p_good=[]
+        a_good=[]
+        r_good=[]
+
+        for p_a_r in zip(pi_files, arf_files, rmf_files):
+            ondisk = [ os.path.exists(x) for x in p_a_r]
+            if all(ondisk) == False:
```
Comment (DougBurke) [inline bin/srcflux:1909]: This is better as

    if not all(ondisk):
        ....

File: bin/srcflux [hunk 3]
```diff
@@ -1969,33 +1990,36 @@ def merge_get_counts( infiles ):
...
+
     good = ( np.array( cts["inside_fov"] ) == True )
 
 
     c      = np.array( cts["counts"] )[good]
     b      = np.array( cts["bg_counts"] )[good]
     alpha  = np.array(cts["psffrac"])[good]
     beta   = np.array(cts["bg_psffrac"] )[good]
+    exptime = np.array(cts["exptime"])[good]
     if True == rate:
```
Comment (DougBurke) [inline bin/srcflux:2008]: Another minor change:

    if rate:
        ...

File: bin/srcflux [hunk 4]
```diff
@@ -2010,23 +2034,120 @@ def merge_aprates( cts, rate=False ):
...
+               }
+
+    pp = [C,A_S,Alpha,T_S,E_S,B,A_B,Beta,T_B,E_B]
+    wtaf=np.isfinite(pp).all()
+
+    if not any(good):
+        delme(outfile.name)
+        return badval
+    
+    if wtaf == False:
```
Comment (DougBurke) [inline bin/srcflux:2087]: if not wtaf:

File: bin/srcflux [hunk 5]
```diff
@@ -2590,14 +2724,17 @@ def obi_main(pars):
         obipars["dtffile"] = infile[5]
 
         if False == is_single_obi:
```
Comment (DougBurke) [inline bin/srcflux:2724]: if not is_single_obi:

File: bin/srcflux [hunk 6]
```diff
@@ -2544,7 +2680,11 @@ def process_single_obi( myparams, pars ):
         photfluxfiles = get_fluximage_flux( taskrunner, myparams, at_energy, src, bkg )
 
         # ---
-        taskrunner.run_tasks( processes=myparams.nproc )
+        try:
+            taskrunner.run_tasks( processes=myparams.nproc )
+        except Exception as ee:
+            print(ee)
```
Comment (DougBurke) [inline bin/srcflux:2684]: I assume this was to help track down issues, so worth keeping in?

**Comment by DougBurke** [comment]
I assume the request at https://github.com/cxcsds/ciao-contrib/pull/387#pullrequestreview-441591002 has been made (numbering of items)?

**Comment by DougBurke** [comment]
We're going to need the changelog updating, but that can be done separately

**Pull request merged**